opm/autodiff/SimulatorFullyImplicitBlackoil_impl.hpp

```diff
@@ -483,13 +483,15 @@ namespace Opm
                       const WellMap&    wmap)
         {
             std::vector<int> resv_prod;
-
-            for (int w = 0, nw = wells.number_of_wells; w < nw; ++w) {
-                if (is_resv_prod(wells, w) ||
-                    ((wells.name[w] != 0) &&
-                     is_resv_prod(wmap, wells.name[w], step)))
-                {
-                    resv_prod.push_back(w);
+            if( &wells )
```

Comment:
Will not that lead to a segfault? Perhaps only when you use the reference?

Anyway: I think that the compiler would be allowed to assume that the pointer is nonzero, or perhaps it is undefined behaviour. It makes me very nervous, because it breaks a very fundamental rule: do not dereference a null pointer. Are you sure this is guaranteed to work by the standard? Even if this is correct, it is so surprising that we should not do it like this.